<launch>
  <arg name="usb_port"                default="/dev/ttyUSB0"/>
  <arg name="dxl_baud_rate"           default="4000000"/>
  <arg name="namespace"               default="ankle_exo_frontal"/>
  <arg name="control_mode"            default="velocity_control"/>

  <arg name="use_dynamixel_server"    default="true"/>
  
  <param name="dynamixel_info"          value="$(find ankle_exoskeleton)/config/basic.yaml"/>

  <node name="$(arg namespace)" pkg="ankle_exoskeleton" type="dynamixel_controllers"
        required="true" output="screen" args="$(arg usb_port) $(arg dxl_baud_rate) $(arg control_mode)">

    <param name="use_dynamixel_server"       value="$(arg use_dynamixel_server)"/>
    <rosparam>
      publish_period: 0.0025
      dxl_read_period: 0.0025
    </rosparam>
  </node>
</launch>
